build_buildboard.py

```
@@ -1,144 +1,96 @@
-    return web_path
-
-def save_photo(url, output_path):
-    print 'saving %s' % os.path.basename(output_path)
-    access_token = 'token %s' % GITHUB_ACCESS_TOKEN
-    response = requests.get(url, stream=True, headers={'Authorization': access_token})
-    with open(output_path, 'wb') as outfile:
-        shutil.copyfileobj(response.raw, outfile)
-    return get_photo_path_for_web(output_path)
-
-def process_yaml_file(yaml_file, download_imgs=False):
-    repo_name = os.path.splitext(os.path.basename(yaml_file))[0]
+from names import *
+
+logging.basicConfig(filename='output.log',level=logging.INFO)
+
+parser = argparse.ArgumentParser(description="Top-level flags.")
+parser.add_argument('--local', action='store_true')
+
+env = Environment(loader=PackageLoader('buildboard', 'templates'),
+                    autoescape=select_autoescape(['html', 'xml']))
+
+def get_yaml_path(team_name):
+    return os.path.join(PWD, OUTPUT_DIR_NAME, YAML_DIR_NAME, "%s.yaml" % team_name)
+
+def get_yaml_doc(team_name):
+    yaml_file = get_yaml_path(team_name)
     try:
         with open(yaml_file, 'r') as report_contents:
             doc = yaml.safe_load(report_contents)
-    except yaml.parser.ParserError, e:
-        print 'repo', repo_name, 'contains bad report.yaml file', str(e)
+            return doc
+    except (yaml.parser.ParserError, yaml.scanner.ScannerError), e:
+        logging.error('repo', team_name, 'contains bad report.yaml file', str(e))
+        return
 
-    try:
-        team_photo = doc['team']['picture']
-        if download_imgs:
-            # save team photo and update yaml to hold relative path
-            team_photo_url = get_photo_url(repo_name, team_photo)
-            team_photo_path = save_photo_path(TEAM_PHOTOS_DIR_NAME, repo_name, team_photo)
-            doc['team']['picture'] = save_photo(team_photo_url, team_photo_path)
-
-        else:  # update paths anyway
+def process_yaml_file(team_name):
+    doc = get_yaml_doc(team_name)
+    if doc:
```

What happens if doc is None?
Should you raise an exception?

---

I added an error message. Most of this is handled in get_yaml_doc instead.